### PR TITLE
feat: add exercise D The Odin Project

### PR DESCRIPTION
**Pull request by Sembauke**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

File: curriculum/challenges/english/16-the-odin-project/top-learn-css-foundations-projects/css-foundations-exercise-d.md [hunk 1]
```diff
@@ -7,38 +7,124 @@ dashedName: css-foundations-exercise-d
...
+1. You should give it a `height` twice as big as it's width.
 
 # --hints--
 
-Test 1
+You should have a `width` of `300px` on the `avatar` and `proportioned` class.
 
 ```js
-
+const style = new __helpers.CSSHelp(document).getStyle(`.avatar.proportioned`);
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/16-the-odin-project/top-learn-css-foundations-projects/css-foundations-exercise-d.md:22]:
```suggestion
const style = new __helpers.CSSHelp(document).getStyle(`.avatar.proportioned`) || new __helpers.CSSHelp(document).getStyle(`.proportioned.avatar`);
```

File: curriculum/challenges/english/16-the-odin-project/top-learn-css-foundations-projects/css-foundations-exercise-d.md [hunk 2]
```diff
@@ -7,38 +7,124 @@ dashedName: css-foundations-exercise-d
...
+assert(style?.width === '300px');
 ```
 
-Test 2
+You should have a height of `auto` on the `avatar` and `proportioned` class.
 
 ```js
+const style = new __helpers.CSSHelp(document).getStyle(`.avatar.proportioned`);
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/16-the-odin-project/top-learn-css-foundations-projects/css-foundations-exercise-d.md:29]:
```suggestion
const style = new __helpers.CSSHelp(document).getStyle(`.avatar.proportioned`) || new __helpers.CSSHelp(document).getStyle(`.proportioned.avatar`)
```

File: curriculum/challenges/english/16-the-odin-project/top-learn-css-foundations-projects/css-foundations-exercise-d.md [hunk 3]
```diff
@@ -7,38 +7,124 @@ dashedName: css-foundations-exercise-d
...
 
 ```js
+const style = new __helpers.CSSHelp(document).getStyle(`.avatar.distorted`);
+assert(style?.width === '200px');
+```
+
+You should use a chaining selector on the `avatar` and `distorted` class.
 
+```js
+const style = new __helpers.CSSHelp(document).getStyle(`.avatar.proportioned`);
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/16-the-odin-project/top-learn-css-foundations-projects/css-foundations-exercise-d.md:50]:
```suggestion
const style = new __helpers.CSSHelp(document).getStyle(`.avatar.distorted`) || new __helpers.CSSHelp(document).getStyle(`.distorted.avatar`);
```

This one was also looking for the wrong class.

**Review by naomi-lgbt — DISMISSED** [review]
Since `.avatar.proportioned` and `.proportioned.avatar` both target the same element and are both valid selectors, we should allow both to pass.

**Review by Nirajn2311 — APPROVED** [review]

**Review by moT01 — APPROVED** [review]
LGTM 🎉 I'm gonna give it a merge.

Wondering if we should remove that save button that saves users code to the database on these. They feel more like mini challenges than projects you might need to save your code for.

**Pull request merged**